nutils/function.py

```diff
@@ -3029,7 +3029,8 @@ def partial_derivative( func, arg_key, arg_axes ):
 
   if not isinstance( arg_key, (int, str) ):
     raise ValueError( 'arg_key: expected an int or str, got {!r}'.format( arg_key ) )
-  arg_axes = tuple(arg_axes)
+  if arg_axes is not None:
+    arg_axes = tuple(arg_axes) # evaluate iterator
```

Comment:
You could drop this statement and move `tuple` to where you create `orig_axes`, but I'd actually like that an error is raised here before anything has happened, if something invalid is passed to `arg_axes`. Eases debugging.